### PR TITLE
feat: Add experimental debug override functionality.

### PR DESCRIPTION
**Pull request by pranjal-jately-ld**
**Requirements**

- [x] I have added test coverage for new or changed functionality
- [x] I have followed the repository's [pull request submission guidelines](../blob/master/CONTRIBUTING.md#submitting-pull-requests)
- [ ] I have validated my changes against all supported platform versions

**Describe the solution you've provided**

This PR adds debug override capability to the LaunchDarkly SDK, allowing developers to override flag values during local development and testing without affecting production flag configurations.

Key features:
- Flag Override Interface: New `LDDebugOverride` interface with methods to set, remove, and manage flag overrides.
- Plugin Integration: Extended `LDPlugin` interface with optional `registerDebug` method for plugins to access override functionality.
- Runtime Override Management: Override values take precedence over remote flag values while preserving original flag evaluation logic.
- Testing: Added test coverage for all override methods and edge cases.

**Additional context**

This feature is designed primarily for development and testing workflows where developers need to:
- Test different flag variations without modifying remote configurations.
- Debug applications with specific flag states.
- Validate feature behavior across different flag combinations.

The implementation maintains backward compatibility and follows the existing plugin architecture patterns. Override functionality is only available when explicitly enabled through plugins, ensuring it doesn't impact production performance or security.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -77,6 +82,37 @@ function initialize(env, context, specifiedOptions, platform, extraOptionDefs) {
   const requestor = Requestor(platform, options, environment);
 
   let flags = {};
+  let flagOverrides = {};
+
+  // Central flag store facade - single source of truth for all flag access
+  const flagStore = {
+    get(key) {
+      // Check overrides first, then real flags
+      if (flagOverrides && utils.objectHasOwnProperty(flagOverrides, key) && flagOverrides[key]) {
```
[the remark this answers is not in the document]
Comment (kinyoklion) [reply src/index.js]: I think most other properties would not influence the application in the same way that a deleted flag would. Maybe this is something we can defer for now and just let overrides be set.

File: src/FlagStore.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
+const utils = require('./utils');
+
+function FlagStore() {
+  let flags = {};
+  let flagOverrides;
```
Comment (kinyoklion) [inline src/FlagStore.js:5]:
```suggestion
  // The flag overrides are set lazily to allow bypassing property checks when no overrides are present.
  let flagOverrides;
```

File: src/FlagStore.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
+const utils = require('./utils');
+
+function FlagStore() {
```
Comment (kinyoklion) [inline src/FlagStore.js:3]: I would suggest a comment block on the FlagStore to explain its function.

**Review by kinyoklion — COMMENTED** [review]
I think things are looking pretty good. I added some minor comments and I have two overall comments. 

I think we maybe want to be able to experiment with this interface. So I would add a comment to registerDebug to indicate that it is experimental, and for use by LaunchDarkly tools at this time.

I would also update the title of the PR, which will become a release note.

`feat: Add experimental debug override functionality.`

I would also like if you could add your updates to the plugin interface to our specification: https://github.com/launchdarkly/sdk-specs/tree/main/specs/PLUGIN-sdk-plugin-support

I think we could tag this as being client-side only for the time being.

**Review by kinyoklion — APPROVED** [review]
